Build fails when depending on `winit` with `android-game-activity` feature enabled

**Describe the bug**
[This example](https://github.com/metent/rust-android-examples/tree/0beec1d0b8c2f3311852b08318d3a7d31a404c84/agdk-eframe) fails to build with latest egui.

**To Reproduce**
Steps to reproduce the behavior:
1. Clone [this repository](https://github.com/metent/rust-android-examples).
2. Navigate to `agdk-eframe` directory.
3. Follow the instructions in `README.md` to build the example.
4. Build fails with the following error.

```
[2023-02-12T06:45:41Z INFO  cargo_ndk::cli] Using NDK at path: /home/rishabh/Android/Sdk/ndk/25.2.9519653 (ANDROID_NDK_HOME)
[2023-02-12T06:45:41Z INFO  cargo_ndk::cli] NDK API level: 21
[2023-02-12T06:45:41Z INFO  cargo_ndk::cli] Building targets: arm64-v8a
[2023-02-12T06:45:41Z INFO  cargo_ndk::cli] Building arm64-v8a (aarch64-linux-android)
warning: file `/home/rishabh/software/myprogs/rust-android-examples/agdk-eframe/src/lib.rs` found to be present in multiple build targets:
  * `lib` target `main`
  * `bin` target `agdk-eframe`
    Updating git repository `https://github.com/emilk/egui`
   Compiling ecolor v0.21.0 (https://github.com/emilk/egui?branch=master#530e9f66)
   Compiling emath v0.21.0 (https://github.com/emilk/egui?branch=master#530e9f66)
   Compiling android-activity v0.4.0
error[E0252]: the name `activity_impl` is defined multiple times
   --> /home/rishabh/.cargo/registry/src/github.com-1ecc6299db9ec823/android-activity-0.4.0/src/lib.rs:107:5
    |
102 | use native_activity as activity_impl;
    |     -------------------------------- previous import of the module `activity_impl` here
...
107 | use game_activity as activity_impl;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `activity_impl` reimported here
    |
    = note: `activity_impl` must be defined only once in the type namespace of this module
help: you can use `as` to change the binding name of the import
    |
107 | use game_activity as other_activity_impl;
    |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error: The "game-activity" and "native-activity" features cannot be enabled at the same time
  --> /home/rishabh/.cargo/registry/src/github.com-1ecc6299db9ec823/android-activity-0.4.0/src/lib.rs:74:1
   |
74 | / compile_error!(
75 | |     "The \"game-activity\" and \"native-activity\" features cannot be enabled at the same time"
76 | | );
   | |_^

   Compiling epaint v0.21.0 (https://github.com/emilk/egui?branch=master#530e9f66)
error[E0308]: mismatched types
   --> /home/rishabh/.cargo/registry/src/github.com-1ecc6299db9ec823/android-activity-0.4.0/src/game_activity/mod.rs:127:41
    |
127 |               inner: Arc::new(RwLock::new(AndroidAppInner {
    |  _____________________________-----------_^
    | |                             |
    | |                             arguments to this function are incorrect
128 | |                 native_app: NativeAppGlue { ptr },
129 | |                 config: ConfigurationRef::new(config),
130 | |                 native_window: Default::default(),
131 | |             })),
    | |_____________^ expected struct `native_activity::AndroidAppInner`, found struct `game_activity::AndroidAppInner`
    |
    = note: struct `game_activity::AndroidAppInner` and struct `native_activity::AndroidAppInner` have similar names, but are actually distinct types
note: struct `game_activity::AndroidAppInner` is defined in module `crate::game_activity` of the current crate
   --> /home/rishabh/.cargo/registry/src/github.com-1ecc6299db9ec823/android-activity-0.4.0/src/game_activity/mod.rs:151:1
    |
151 | pub struct AndroidAppInner {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^
note: struct `native_activity::AndroidAppInner` is defined in module `crate::native_activity` of the current crate
   --> /home/rishabh/.cargo/registry/src/github.com-1ecc6299db9ec823/android-activity-0.4.0/src/native_activity/mod.rs:127:1
    |
127 | pub(crate) struct AndroidAppInner {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: associated function defined here
   --> /home/rishabh/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/sync/rwlock.rs:160:18
    |
160 |     pub const fn new(t: T) -> RwLock<T> {
    |                  ^^^

error[E0308]: mismatched types
   --> /home/rishabh/.cargo/registry/src/github.com-1ecc6299db9ec823/android-activity-0.4.0/src/game_activity/mod.rs:251:49
    |
251 | ...                   loader: StateLoader { app: self },
    |                               ^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `native_activity::StateLoader`, found struct `game_activity::StateLoader`
    |
    = note: struct `game_activity::StateLoader` and struct `native_activity::StateLoader` have similar names, but are actually distinct types
note: struct `game_activity::StateLoader` is defined in module `crate::game_activity` of the current crate
   --> /home/rishabh/.cargo/registry/src/github.com-1ecc6299db9ec823/android-activity-0.4.0/src/game_activity/mod.rs:80:1
    |
80  | pub struct StateLoader<'a> {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^
note: struct `native_activity::StateLoader` is defined in module `crate::native_activity` of the current crate
   --> /home/rishabh/.cargo/registry/src/github.com-1ecc6299db9ec823/android-activity-0.4.0/src/native_activity/mod.rs:49:1
    |
49  | pub struct StateLoader<'a> {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
   --> /home/rishabh/.cargo/registry/src/github.com-1ecc6299db9ec823/android-activity-0.4.0/src/game_activity/mod.rs:255:52
    |
255 | ...                   saver: StateSaver { app: self },
    |                              ^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `native_activity::StateSaver`, found struct `game_activity::StateSaver`
    |
    = note: struct `game_activity::StateSaver` and struct `native_activity::StateSaver` have similar names, but are actually distinct types
note: struct `game_activity::StateSaver` is defined in module `crate::game_activity` of the current crate
   --> /home/rishabh/.cargo/registry/src/github.com-1ecc6299db9ec823/android-activity-0.4.0/src/game_activity/mod.rs:39:1
    |
39  | pub struct StateSaver<'a> {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^
note: struct `native_activity::StateSaver` is defined in module `crate::native_activity` of the current crate
   --> /home/rishabh/.cargo/registry/src/github.com-1ecc6299db9ec823/android-activity-0.4.0/src/native_activity/mod.rs:33:1
    |
33  | pub struct StateSaver<'a> {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^

Some errors have detailed explanations: E0252, E0308.
For more information about an error, try `rustc --explain E0252`.
error: could not compile `android-activity` due to 5 previous errors
warning: build failed, waiting for other jobs to finish...
[2023-02-12T06:45:54Z INFO  cargo_ndk::cli] If the build failed due to a missing target, you can run this command:
[2023-02-12T06:45:54Z INFO  cargo_ndk::cli] 
[2023-02-12T06:45:54Z INFO  cargo_ndk::cli]     rustup target install aarch64-linux-android
```

**Expected behavior**
The example should build successfully.

**Desktop (please complete the following information):**
 - OS: Linux
 - Version: master branch

**Additional context**
According to the error log, the build probably fails because two conflicting features `native-activity` and `game-activity` are enabled at the same time. `android-game-activity` feature is enabled for `winit` dependency in the example and `native-activity` is enabled by `egui-winit`. The example builds successfully if [this line](https://github.com/emilk/egui/blob/530e9f667c0969b09edb68f7701f20c52a273d3a/crates/egui-winit/Cargo.toml#L76) is removed.

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -199,7 +199,7 @@ jobs:
       - name: Set up cargo cache
         uses: Swatinem/rust-cache@v2
 
-      - run: cargo check --features wgpu --target aarch64-linux-android
+      - run: cargo check --features wgpu,android-native-activity --target aarch64-linux-android
         working-directory: crates/eframe
 
   # ---------------------------------------------------------------------------

--- a/crates/eframe/Cargo.toml
+++ b/crates/eframe/Cargo.toml
@@ -69,6 +69,15 @@ __screenshot = ["dep:image"]
 ## This overrides the `glow` feature.
 wgpu = ["dep:wgpu", "dep:egui-wgpu", "dep:pollster"]
 
+# Allow crates to choose an android-activity backend via Winit
+# - It's important that most applications should not have to depend on android-activity directly, and can
+#   rely on Winit to pull in a suitable version (unlike most Rust crates, any version conflicts won't link)
+# - It's also important that we don't impose an android-activity backend by taking this choice away from applications.
+
+## Enable the `native-activity` backend via `egui-winit` on Android
+android-native-activity = [ "egui-winit/android-native-activity" ]
+## Enable the `game-activity` backend via `egui-winit` on Android
+android-game-activity = [ "egui-winit/android-game-activity" ]
 
 [dependencies]
 egui = { version = "0.21.0", path = "../egui", default-features = false, features = [

--- a/crates/egui-winit/Cargo.toml
+++ b/crates/egui-winit/Cargo.toml
@@ -42,6 +42,16 @@ serde = ["egui/serde", "dep:serde"]
 ## Enables Wayland support.
 wayland = ["winit/wayland"]
 
+# Allow crates to choose an android-activity backend via Winit
+# - It's important that most applications should not have to depend on android-activity directly, and can
+#   rely on Winit to pull in a suitable version (unlike most Rust crates, any version conflicts won't link)
+# - It's also important that we don't impose an android-activity backend by taking this choice away from applications.
+
+## Enable the `native-activity` backend via Winit on Android
+android-native-activity = [ "winit/android-native-activity" ]
+## Enable the `game-activity` backend via Winit on Android
+android-game-activity = [ "winit/android-game-activity" ]
+
 [dependencies]
 egui = { version = "0.21.0", path = "../egui", default-features = false, features = [
   "tracing",
@@ -76,6 +86,3 @@ smithay-clipboard = { version = "0.6.3", optional = true }
 [target.'cfg(not(target_os = "android"))'.dependencies]
 arboard = { version = "3.2", optional = true, default-features = false }
 
-[target.'cfg(target_os = "android")'.dependencies]
-# TODO(emilk): this is probably not the right place for specifying native-activity, but we need to do it somewhere for the CI
-android-activity = { version = "0.4", features = ["native-activity"] }

